### PR TITLE
[FLINK-33013][connectors/mongodb] Shade flink-connector-base into flink-sql-connector-mongodb

### PR DESCRIPTION
**Pull request by Jiabao-Sun**
Shade `flink-connector-base` into `flink-sql-connector-mongodb`

### CODE REVIEW

**Comment by Jiabao-Sun** [comment]
Hi @leonardBang, could you help review this when you have time?

File: flink-sql-connector-mongodb/pom.xml [hunk 1]
```diff
@@ -60,6 +60,10 @@ under the License.
 							<artifactSet>
 								<includes>
 									<include>*:*</include>
+									<include>org.apache.flink:flink-connector-base</include>
+									<include>org.apache.flink:flink-connector-mongodb</include>
+									<include>org.mongodb:*</include>
+									<include>org.bson:*</include>
```
Comment (ruanhang1993) [inline flink-sql-connector-mongodb/pom.xml:66]: I think these new includes are useless as `<include>*:*</include>` already contains them.

Comment (Jiabao-Sun) [reply flink-sql-connector-mongodb/pom.xml]: Thanks @ruanhang1993 to remind me of this.
The old shaded package does not include `flink-connector-base` is that has provided scope.
I removed the useless includes and it works.
Please help review it again when you have time.

Comment (Jiabao-Sun) [reply flink-sql-connector-mongodb/pom.xml]: Thanks @ruanhang1993 to remind me of this.
The old shaded package does not include `flink-connector-base` is that has provided scope.
I removed the useless includes and it works.
Please help review it again when you have time.

**Review by ruanhang1993 — APPROVED** [review]
LGTM.

**Comment by Jiabao-Sun** [comment]
As the [Externalized Connector development](https://cwiki.apache.org/confluence/display/FLINK/Externalized+Connector+development) says.
> Bundling of flink-connector-base
Connectors should not  bundle the connector-base module from Flink and instead set it to provided, as contained classes may rely on internal Flink classes.

The `flink-connector-base` has been put into recent fink dist lib.
Closed this issue because we needn't shaded `flink-connector-base` into connectors any more.

**Pull request closed without merging**